include/openssl/ssl.h

```diff
@@ -4886,6 +4886,11 @@ OPENSSL_EXPORT int SSL_CTX_set1_sigalgs_list(SSL_CTX *ctx, const char *str);
 // more convenient to codesearch for specific algorithm values.
 OPENSSL_EXPORT int SSL_set1_sigalgs_list(SSL *ssl, const char *str);
 
+// SSL_CTX_get_security_level returns 3. This is only to maintain compatibility
```

Comment:
Do all of our SSL configurations meet OpenSSL's definition of level 3 security?

Reply:
Yeah I was on the edge on documenting this, but I was worried that entire description might make things a bit bloated. Here's what OpenSSL's definition of level 2 and 3 is:
>Level 2
Security level set to 112 bits of security. As a result RSA, DSA and DH keys shorter than 2048 bits and ECC keys shorter than 224 bits are prohibited. In addition to the level 1 exclusions any cipher suite using RC4 is also prohibited. SSL version 3 is also not allowed. Compression is disabled.

 > Level 3
Security level set to 128 bits of security. As a result RSA, DSA and DH keys shorter than 3072 bits and ECC keys shorter than 256 bits are prohibited. In addition to the level 2 exclusions cipher suites not offering forward secrecy are prohibited. TLS versions below 1.1 are not permitted. Session tickets are disabled.

We only support 128 bits of security, but we don't seem to prohibit session tickets and TLS 1.0 like Level 3 states. But reporting Level 2 seems to be even more out of line since we don't support 112 bits.

Reply:
I like the extra clarity. The question remains what if someone configs a connection with 256 bit cipher suites and expects this function to return 5 for them?

Reply:
I think this function only returns the "current security level" of the library. It seems to be based on how the library is configured, instead of per connection. Per the documentation:
> The default security level changes when OpenSSL is compiled by `DOPENSSL_TLS_SECURITY_LEVEL=level` ...

Or when setting the security level with "The functions SSL_CTX_set_security_level() and SSL_set_security_level() set the security level to level. If not set the library default security level is used."

We don't support setting security levels this way yet, so I think `3` is the most suitable number to return for our existing cipher suites now. But your question does address that we'll have to change this behavior once we get an ask to support actual "security_level" setting in AWS-LC.

Reply:
Thanks, that clears this up, I agree 3 makes sense.